Bug/clean spec definitions section of spec to avoid confusion (#156)
asyncapi/asyncapi: versions/next/asyncapi.md

The AsyncAPI Specification defines a set of files required to describe such an API.
These files can then be used to create utilities, such as documentation, integration and/or testing tools.

> The file(s) MUST describe the operations a new [application](#definitionsApplication) can perform. For instance:

Reviewer: What do you think about changing the point of view here?

For instance:
```suggestion
The file(s) MUST describe the operations an [application](#definitionsApplication) performs. For instance:
```

So, instead of looking at it from the consumer/client perspective, we use the AsyncAPI document to define what the application actually does.

Author: Agreed, will fix

```yaml
user/signedup:
  subscribe:
    $ref: "#/components/messages/userSignUp"
```

> It means [application](#definitionsApplication) can subscribe to `user/signedup` channel. However, it does NOT mean every [application](#definitionsApplication) must subscribe to this channel.

Reviewer: So, following with the same reasoning, since an AsyncAPI document represents a single application, this sentence doesn't make sense anymore.

For context:
I wrote this sentence thinking about people who wanted to use AsyncAPI to document their whole event-driven microservices architecture. Not just a single service (application). But I realize it introduces more problems. That's why I suggested this feature: https://github.com/asyncapi/asyncapi/issues/146. So we make it clear the document represents the application.

Author: Agreed will clean up

## Definitions

#### <a name="definitionsMessageBroker"></a>Message Broker
A message broker is an intermediary component that translates a [message](#definitionsMessage) from the messaging scheme of the sender to the messaging scheme of the receiver.  The message broker mediates interactions between applications, minimizing the mutual awareness that applications should have of each other in order to be able to exchange messages, effectively providing decoupling. A message broker MUST support at least one [scheme](#definitionsScheme), thus enabling message exchanges between applications.  Importantly, the message broker is a mechanism that enables asynchronous application patterns such as publish/subscribe, fire and forget, request/reply and queuing. Additionally, the message broker MAY provide additional features such as persistence, storage/replay, authentication/authorization, scheme transformation and message integrity.

#### <a name="definitionsMessage"></a>Message

> A message is the mechanism by which information is exchanged via a channel between [message brokers](#definitionsMessageBroker) and applications. A message contains a header and a payload. The header MAY be subdivided into [scheme](#definitionsScheme) defined headers and header properties defined by the application which can act as supporting metadata. The payload contains the data, defined by the application, which MUST be serialized into a formats (JSON, XML, Avro, binary, etc). Since a message is a generic mechanism, it can support multiple interaction patterns such as event, command, request, or response. 

Reviewer: ```suggestion
A message is the mechanism by which information is exchanged via a channel between [message brokers](#definitionsMessageBroker) and applications. A message MUST contain a payload and MAY also contain a header. The header MAY be subdivided into [scheme](#definitionsScheme) defined headers and header properties defined by the application which can act as supporting metadata. The payload contains the data, defined by the application, which MUST be serialized into a format (JSON, XML, Avro, binary, etc.). Since a message is a generic mechanism, it can support multiple interaction patterns such as event, command, request, or response. 
```

#### <a name="definitionsChannel"></a>Channel
A channel is an addressable component, made available by the [message broker](#definitionsMessageBroker), for the organization of [messages](#definitionsMessage). [Producer](#definitionsProducer) applications send messages to channels and [consumer](#definitionsConsumer) applications consume messages from Channels. Message Brokers support many channel instances, allowing messages with different content to be addressed to different channels. Message Brokers MAY provide multiple channel types thus enabling different asynchronous application patterns such as topics, queues and exchanges. Depending on the message broker implementation, the channel MAY be included in the message via scheme defined headers.

#### <a name="definitionsScheme"></a>Scheme

> A scheme is the mechanism (wireline protocol OR API) by which [messages](#definitionsMessage) are exchanged between the application and the [channel](#definitionsChannel) hosted by the [message broker](#definitionsMessageBroker). Example schemes include, but are not limited to, AMQP, MQTT, WebSocket, JMS, Kafka, STOMP.  

Reviewer: ```suggestion
A scheme is the mechanism (wireline protocol or API) by which [messages](#definitionsMessage) are exchanged between the application and the [channel](#definitionsChannel) hosted by the [message broker](#definitionsMessageBroker). Example schemes include, but are not limited to, AMQP, MQTT, WebSocket, JMS, Kafka, STOMP.  
```

#### <a name="definitionsApplication"></a>Application

> An application is any kind of computer program connected to a [message broker](#definitionsMessageBroker). It MUST be a [producer](#definitionsProducer), a [consumer](#definitionsConsumer) or both. An application MAY be a microservice, IoT device (sensor), mainframe process, etc. An application MAY be written in any number of different programming languages as long as its supported by the selected [scheme](#definitionsScheme). An application MUST also use a scheme supported by the message broker in order to connect and exchange [messages](#definitionsMessage). 

Reviewer: ```suggestion
An application is any kind of computer program connected to a [message broker](#definitionsMessageBroker). It MUST be a [producer](#definitionsProducer), a [consumer](#definitionsConsumer) or both. An application MAY be a microservice, IoT device (sensor), mainframe process, etc. An application MAY be written in any number of different programming languages as long as they support the selected [scheme](#definitionsScheme). An application MUST also use a scheme supported by the message broker in order to connect and exchange [messages](#definitionsMessage). 
```
